parsers/marburg.py

```diff
@@ -1,61 +1,73 @@
 from urllib.request import urlopen
 from bs4 import BeautifulSoup as parse
-from bs4.element import Tag
 import re
 
 from utils import Parser
 
-from pyopenmensa.feed import LazyBuilder, extractWeekDates
-
-employeePrice = re.compile('Unibedienstetenzuschlag:? ?(?P<price>\d+[,.]\d{2}) ?€?')
-otherPrice = re.compile('Gästezuschlag:? ?(?P<price>\d+[,.]\d{2}) ?€?')
-
-
-def parse_week(url, canteen, mensa):
-    document = parse(urlopen(url).read())
-    # extra legends information
-    canteen.setLegendData(text=document.find(text='Kennzeichnung: ').parent.next_sibling.get_text().replace('&nbsp;', ' '))
-    # additional charges
-    prices = {}
-    for p in document.find_all('p'):
-        match = employeePrice.search(p.text)
-        if match:
-            prices['employee'] = match.group('price')
-        match = otherPrice.search(p.text)
-        if match:
-            prices['other'] = match.group('price')
-    if len(prices) != 2:
-        print('Could not extract addtional charges for employee and others')
-    canteen.setAdditionalCharges('student', prices)
-    # find
-    mensa_data = document.find('h1', text=re.compile(mensa)).parent
-    while type(mensa_data) != Tag or mensa_data.name != 'div'\
-            or 'tx-cagcafeteria-pi1' not in mensa_data.get('class', []):
-        mensa_data = mensa_data.next_sibling
-    weekDays = extractWeekDates(mensa_data.find('h2').text)
-    for day_headline in mensa_data.find_all('h3'):
-        date = weekDays[day_headline.text]
-        day_table = day_headline.next_sibling.next_sibling
-        for tr_menu in day_table.tbody.find_all('tr'):
-            category = tr_menu.find_all('td')[0].text.strip()
-            name = tr_menu.find_all('td')[1].text.replace('\r\n', ' ').strip()
-            canteen.addMeal(date, category, name, [], tr_menu.find_all('td')[2].text)
-
-
-def parse_url(url, mensa, *weeks, today):
+from pyopenmensa.feed import LazyBuilder, extractDate
+
+heading_regex = re.compile(r'^h[1-6]$')
+notes_regex = re.compile(r'\((\d+,?\s*)*\)\s*')
+price_regex = re.compile(r'(?P<price>\d+[,.]\d{2}) ?€')
+
+roles = ('student',)
+
+
+def parse_dish(dish, canteen):
+    date = extractDate(dish['data-date'])
+
+    name = dish.find(class_='neo-menu-single-title')
+    try:
+        notes = set(x['title'] for x in name.find_all(name='abbr'))
+    except AttributeError:
+        # No Name
+        return
+    name = re.sub(notes_regex, '', name.get_text(strip=True))
```

Comment:
You could also use `name.text.strip()`, which is shorter. :wink:

Also you should check if the name is empty afterwards, because empty meal names are not allowed and result in: `ValueError: Meal names must not be empty`